### PR TITLE
Subview toggle helper

### PR DESCRIPTION
**Pull request by JanX2**
This is based off of the Swift 3 branch.

The MDPSplitView+SubviewToggleHelper category is a generalized and refactored Obj-C implementation of the toggle method from “MainWindowController.swift”.

### CODE REVIEW

**Comment by mdiep** [comment]
Can you merge `master` into here so that the diff is updated? 🙏

**Comment by JanX2** [comment]
I can rebase. That’s cleaner I think. :)

**Comment by JanX2** [comment]
There.

**Review by mdiep — CHANGES_REQUESTED** [review]
I think it makes sense to include these helper methods, but I'd like to see a few changes:

1. Move these into the existing source files
2. Add full documentation for the methods
3. Match the existing style and indentation

(The xcodeproj is showing that it's configured for tabs for some reason, but the files are indented with spaces. Spaces are the desired indentation.)

**Comment by JanX2** [comment]
Will do. Friday at the earliest. Probably early next week. :)

**Comment by JanX2** [comment]
Done.

File: MDPSplitView.h [hunk 1]
```diff
@@ -58,4 +61,70 @@ FOUNDATION_EXPORT const unsigned char MDPSplitViewVersionString[];
...
+ @param subview
+ The subview to toggle.
+ @param dividerIndex
+ The index of the divider to position.
+ @param lastWidth
+ A reference to the variable you store the last width of the subview in.
+ @param animationDuration
+ The duration of the animation.
+ @param collapsesRightward
+ Whether you want to have the subview collapse rightward (or leftward, if NO).
```
Comment (mdiep) [inline MDPSplitView.h]: Can you indent the descriptions? (To match the style above)

**Review by mdiep — CHANGES_REQUESTED** [review]

**Comment by JanX2** [comment]
Done.

BTW: I back-ported the Demo to Obj-C, if you ever are interested in that. I will not change anything style-related there, though. :D

**Review by mdiep — APPROVED** [review]

**Comment by mdiep** [comment]
Thanks!

**Pull request merged**